standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/Warehouse.java

```diff
@@ -500,8 +501,16 @@ public void recycleDirToCmPath(Path f, boolean ifPurge) throws MetaException {
   }
 
   public boolean isEmptyDir(Path path) throws IOException, MetaException {
+    return isEmptyDir(path, null);
```

Comment:
Can you pass `p -> true` instead of `null` ?

Also it seems that `FileSystem.listStatus` has a bigger memory footprint when a filter is passed:
https://github.com/apache/hadoop/blob/a3b9c37a397ad4188041dd80621bdeefc46885f2/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java#L2011